src/test/shared/featureController.test.ts

```diff
@@ -0,0 +1,73 @@
+/*!
+ * Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import * as assert from 'assert'
+import { FeatureController } from '../../shared/featureController'
+import { TestSettingsConfiguration } from '../utilities/testSettingsConfiguration'
+
+describe('FeatureController', async () => {
+
+    const testConfig = new TestSettingsConfiguration()
+    const activeSettingKey = 'activeSetting'
+    const inactiveSettingKey = 'inactiveSetting'
+    const missingSettingKey = 'missingSetting'
+    const variableSettingKey = 'variableSetting'
+    const permanentSettingKey = 'permanentSetting'
+    const permanentMissingSettingKey = 'permanentMissingSetting'
+    const permanentMissingVariableSettingKey = 'permanentMissingVariableSetting'
+    await testConfig.writeSetting(`toggle.${activeSettingKey}`, true)
+    await testConfig.writeSetting(`toggle.${inactiveSettingKey}`, false)
+    await testConfig.writeSetting(`toggle.${variableSettingKey}`, false)
+    await testConfig.writeSetting(`toggle.${permanentSettingKey}`, true)
+
+    const features = new FeatureController(testConfig, [
+        permanentSettingKey,
+        permanentMissingSettingKey,
+        permanentMissingVariableSettingKey
+    ])
+
+    it('returns whether or not a feature is active', () => {
```

Comment:
Ideally each test checks for one thing. Tests like this can be split into two...
```
describe FeatureController
    describe isFeatureActive
        it "returns true for active feature"
        it "returns false for non-active feature"

Reply:
✔️